src/wp-includes/class-wpdb.php

```diff
@@ -1553,106 +1555,130 @@ public function prepare( $query, ...$args ) {
 
 		// If args were passed as an array, as in vsprintf(), move them up.
 		$passed_as_array = ( isset( $args[0] ) && is_array( $args[0] ) && 1 === count( $args ) );
+		if ( $passed_as_array && isset( $split_query[2] ) && substr( $split_query[2], 1, 3 ) === '...' && isset( $args[0][0] ) && false === is_array( $args[0][0] ) ) {
+			$passed_as_array = false; // The first (and only) placeholder is using variadics (e.g. '%...d'), and that array has *not* been $passed_as_array, e.g. `$wpdb->prepare('id IN (%...d)', [ 1, 2, 3 ] );`.
```

Comment:
```suggestion
			/*
			 * The first (and only) placeholder is using variadics (e.g. '%...d'),
			 * and that array has *not* been $passed_as_array.
			 *
			 * e.g. `$wpdb->prepare('id IN (%...d)', [ 1, 2, 3 ] );`.
			 */
			$passed_as_array = false;
```
Let's make this a multiline comment to save on horizontal scrolling. 🙂